src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -0,0 +1,514 @@
+        PrimitiveType.PrimitiveTypeName typeName =
+            columnDescriptor.getPrimitiveType().getPrimitiveTypeName();
+        ColumnMetadata column = fieldIndex.get(columnName).getFirst();
+        ColumnLogicalType logicalType = ColumnLogicalType.valueOf(column.getLogicalType());
+        switch (typeName) {
+          case BOOLEAN:
+            int intValue = DataValidationUtil.validateAndParseBoolean(value);
+            value = intValue > 0;
+            stats.addIntValue(BigInteger.valueOf(intValue));
+            size += 1;
+            break;
+          case INT32:
+            int intVal = getInt32Val(value, column, logicalType);
+            value = intVal;
+            stats.addIntValue(BigInteger.valueOf(intVal));
+            size += 4;
+            break;
+          case INT64:
+            long longValue = getInt64Value(value, column, logicalType);
+            value = getInt64Value(value, column, logicalType);
+            stats.addIntValue(BigInteger.valueOf(longValue));
+            size += 8;
+            break;
+          case DOUBLE:
+            double doubleValue = DataValidationUtil.validateAndParseReal(value);
+            value = doubleValue;
+            stats.addRealValue(doubleValue);
+            size += 8;
+            break;
+          case BINARY:
+            String str;
+            if (logicalType.isObject()) {
+              str =
+                  logicalType == ColumnLogicalType.OBJECT
+                      ? DataValidationUtil.validateAndParseObject(value)
+                      : DataValidationUtil.validateAndParseVariant(value);
+            } else {
+              String maxLengthString = column.getLength().toString();
+              str =
+                  DataValidationUtil.validateAndParseString(
+                      value,
+                      Optional.of(maxLengthString)
+                          .map(s -> DataValidationUtil.validateAndParseInteger(maxLengthString)));
+              stats.addStrValue(str);
+            }
+            value = str;
+            size += str.getBytes().length;
+            break;
+          case FIXED_LEN_BYTE_ARRAY:
+            switch (logicalType) {
+              case TIMESTAMP_LTZ:
+              case TIMESTAMP_NTZ:
+                value =
+                    DataValidationUtil.validateAndParseTimestampNtzSb16(value, column.getScale())
+                        .getTimeInScale();
+                break;
+              case TIMESTAMP_TZ:
+                value =
+                    DataValidationUtil.validateAndParseTimestampTz(value, column.getScale())
+                        .getTimeInScale();
+                break;
+              default:
+                break;
+            }
+            value = getSb16Bytes(parseSb16(value, stats, column.getScale()));
+            size += 16;
+            break;
+          default:
+            throw new ParquetEncodingException("Unsupported column type: " + typeName);
```

Comment:
why we're throwing an encoding exception for non-encoding operations?

Reply:
indeed changed:
```
throw new SFException(ErrorCode.UNKNOWN_DATA_TYPE, logicalType, physicalType);
```